packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -140,7 +140,15 @@ abstract class Route<T> {
   ///
   /// If the [settings] are not provided, an empty [RouteSettings] object is
   /// used instead.
-  Route({ RouteSettings? settings }) : _settings = settings ?? const RouteSettings();
+  Route({ RouteSettings? settings }) : _settings = settings ?? const RouteSettings() {
+    if (kFlutterMemoryAllocationsEnabled) {
```

Comment:
Sorry I have not brought up this earlier. Should these logic be wrapped in an assert?

Reply:
yes, thanks for explanation